tests/main/lxd/task.yaml

```diff
@@ -94,7 +94,7 @@ execute: |
     echo "Ensure we can use snapd inside lxd"
     lxd.lxc exec my-ubuntu snap install test-snapd-tools
     echo "And we can run snaps as regular users"
-    lxd.lxc exec my-ubuntu -- su -c "/snap/bin/test-snapd-tools.echo from-the-inside" ubuntu | MATCH from-the-inside
+    lxd.lxc exec my-ubuntu -- su -c "cd /home/ubuntu && /snap/bin/test-snapd-tools.echo from-the-inside" ubuntu | MATCH from-the-inside
```

Comment:
You mentioned that you changed this but it seems like a valid use case for the root user to be in /root and to run something under a shell as non-root. Put another way, does this PR introduce a regression/limitation or was this just a simple cleanup along the way? IIUC, DAC prevented the cd to /root. Perhaps in case of EACCES we go to the void directory rather than dying?

Reply:
Yes it does.

In the past you would be able to move to a directory you don't have permission to enter. Specifically snap-confine would allow you to go into /root despite the permissions on this.

Reply:
We can consider moving to void but the point is that in the past you were able to abuse this power.

Reply:
Well, yes, my point is that perhaps we don't want to break people-- moving to void would at least allow the command to continue for this case that is actually legitimate.

Reply:
Actually, here I would argue that su is at fault, allowing us to stay in /root without having permissions to do so.

Reply:
I agree with @jdstrand if there's a way to keep it working without security issues it's better, we have been bitten already a few times by taking stuff away from people

Reply:
I've change it to use the void directory. I will push this after a round of local testing.

Reply:
Fixed now.